src/umockcallrecorder.c

```diff
@@ -163,26 +163,42 @@ int umockcallrecorder_add_actual_call(UMOCKCALLRECORDER_HANDLE umock_call_record
         /* Codes_SRS_UMOCK_C_LIB_01_115: [ umock_c shall compare calls in order. ]*/
         for (i = 0; i < umock_call_recorder->expected_call_count; i++)
         {
-            if (umock_call_recorder->expected_calls[i].is_matched == 0)
+            int ignore_all_calls = umockcall_get_ignore_all_calls(umock_call_recorder->expected_calls[i].umockcall);
+            if (ignore_all_calls < 0)
             {
-                /* Codes_SRS_UMOCKCALLRECORDER_01_017: [ Comparing the calls shall be done by calling umockcall_are_equal. ]*/
-                int are_equal_result = umockcall_are_equal(umock_call_recorder->expected_calls[i].umockcall, mock_call);
-                if (are_equal_result == 1)
-                {
-                    /* Codes_SRS_UMOCKCALLRECORDER_01_016: [ If the call matches one of the expected calls, a handle to the matched call shall be filled into the matched_call argument. ]*/
-                    *matched_call = umock_call_recorder->expected_calls[i].umockcall;
-                    break;
-                }
-                /* Codes_SRS_UMOCKCALLRECORDER_01_021: [ If umockcall_are_equal fails, umockcallrecorder_add_actual_call shall fail and return a non-zero value. ]*/
-                else if (are_equal_result != 0)
-                {
-                    is_error = 1;
-                    break;
-                }
-                else
+                /* Codes_SRS_UMOCKCALLRECORDER_01_058: [ If getting `ignore_all_calls` by calling `umockcall_get_ignore_all_calls` fails, `umockcallrecorder_add_actual_call` shall fail and return a non-zero value. ]*/
+                UMOCK_LOG("umockcallrecorder: Cannot get the ignore_all_calls flag.");
+                is_error = 1;
+                break;
+            }
+            else
+            {
+                if ((umock_call_recorder->expected_calls[i].is_matched == 0) ||
+                    /* Codes_SRS_UMOCKCALLRECORDER_01_057: [ If any expected call has `ignore_all_calls` set and the actual call is equal to it when comparing the 2 calls, then the call shall be considered matched and not added to the actual calls list. ]*/
+                    (ignore_all_calls == 1))
```

Comment:
Sure.

---
In reply to: [106218840](https://github.com/Azure/umock-c/pull/2#discussion_r106218840) [](ancestors = 106218840)